### PR TITLE
UCT/CMA: Add stub cap for memory invalidation

### PR DESCRIPTION
**Pull request by brminich**
## What
Add en var for enabling "stub" memory invalidation on CMA

## Why ?
For CMA to be selected for RNDV protocol when error handling is requested

before:
```
$UCX_TLS=cma,tcp  ./src/tools/info/ucx_info -e -ute
#
# UCP endpoint
#
#               peer: jazz01:175116
#                 lane[0]:  3:tcp/ib3.0 md[0]               -> md[0]/tcp/sysdev[255] am am_bw#0 wireup
#
#                tag_send: 0..<egr/short>..8185..<egr/zcopy>..975776..<rndv>..(inf)
#            tag_send_nbr: 0..<egr/short>..8185..<egr/bcopy>..262144..<rndv>..(inf)
#           tag_send_sync: 0..<egr/short>..8185..<egr/zcopy>..975776..<rndv>..(inf)
#
#                  rma_bw: mds rndv_rkey_size 9
#
```

after:
```
$UCX_CMA_MEMORY_INVALIDATE=y UCX_TLS=cma,tcp ./src/tools/info/ucx_info -e -ute
#
# UCP endpoint
#
#               peer: jazz01:175081
#                 lane[0]:  3:tcp/ib3.0 md[0]               -> md[0]/tcp/sysdev[255] am am_bw#0 wireup
#                 lane[1]:  6:cma/memory.0 md[1]            -> md[1]/cma/sysdev[255] rma_bw#0
#
#                tag_send: 0..<egr/short>..8185..<egr/zcopy>..770993..<rndv>..(inf)
#            tag_send_nbr: 0..<egr/short>..8185..<egr/bcopy>..262144..<rndv>..(inf)
#           tag_send_sync: 0..<egr/short>..8185..<egr/zcopy>..770993..<rndv>..(inf)
#
#                  rma_bw: mds rndv_rkey_size 9
#
```

### CODE REVIEW

File: src/uct/sm/scopy/cma/cma_md.c [hunk 1]
```diff
@@ -27,6 +27,27 @@
 #endif
 
 
+typedef struct uct_cma_md {
+    struct uct_md       super;
+    uint64_t            extra_caps;
```
Comment (yosefe) [inline src/uct/sm/scopy/cma/cma_md.c:32]: extra_cap_flags

File: src/uct/sm/scopy/cma/cma_md.c [hunk 2]
```diff
@@ -27,6 +27,27 @@
...
+    uct_md_config_t     super;
+    int                 mem_invalidate;
+} uct_cma_md_config_t;
+
+static ucs_config_field_t uct_cma_md_config_table[] = {
+    {"", "", NULL,
+     ucs_offsetof(uct_cma_md_config_t, super),
+     UCS_CONFIG_TYPE_TABLE(uct_md_config_table)},
+
+    {"MEMORY_INVALIDATE", "n", "Expose memory invalidate support capability",
```
Comment (yosefe) [inline src/uct/sm/scopy/cma/cma_md.c:45]: should it be yes/no/auto?

Comment (brminich) [reply src/uct/sm/scopy/cma/cma_md.c:45]: what would "auto" mean?

Comment (yosefe) [reply src/uct/sm/scopy/cma/cma_md.c:45]: it we add cam invalidate later, it will not disable it

Comment (yosefe) [reply src/uct/sm/scopy/cma/cma_md.c:45]: or maybe document that this new config param means we expose invalidate support w/o actually supporting it?

Comment (brminich) [reply src/uct/sm/scopy/cma/cma_md.c:45]: I guess we will have to delete this var when we add real support. 
So, add doc, but leave yes/no?

Comment (yosefe) [reply src/uct/sm/scopy/cma/cma_md.c:45]: ok

**Review by yosefe — APPROVED** [review]

**Comment by brminich** [comment]
@yosefe, port to 1.12?

**Comment by yosefe** [comment]
> @yosefe, port to 1.12?

Yes

**Pull request merged**